extensions/scarb-doc/src/main.rs

```diff
@@ -53,12 +74,135 @@ fn main() -> Result<()> {
 
     for (item_id, doc) in item_documentation.iter() {
         let name = match item_id {
-            LookupItemId::ModuleItem(item_id) => item_id.name(&db),
-            LookupItemId::TraitItem(item_id) => item_id.name(&db),
-            LookupItemId::ImplItem(item_id) => item_id.name(&db),
+            LookupItemId::ModuleItem(item_id) => item_id.name(db),
+            LookupItemId::TraitItem(item_id) => item_id.name(db),
+            LookupItemId::ImplItem(item_id) => item_id.name(db),
         };
         println!("{:?}: {:?} -> {:?}", name, item_id, doc);
     }
 
     Ok(())
 }
+
+fn get_project_config(metadata: &Metadata, package_metadata: &PackageMetadata) -> ProjectConfig {
+    let compilation_unit_metadata =
+        package_lib_compilation_unit(metadata, package_metadata.id.clone())
+            .expect("Failed to find compilation unit for package");
+    let corelib = get_corelib(compilation_unit_metadata);
+    let dependencies = get_dependencies(compilation_unit_metadata);
+    let crates_config = get_crates_config(metadata, compilation_unit_metadata);
+
+    ProjectConfig {
+        base_path: package_metadata.root.clone().into(),
+        corelib: Some(Directory::Real(corelib.source_root().into())),
+        content: ProjectConfigContent {
+            crate_roots: dependencies,
+            crates_config,
+        },
+    }
+}
+
+fn package_lib_compilation_unit(
+    metadata: &Metadata,
+    package_id: PackageId,
+) -> Option<&CompilationUnitMetadata> {
+    metadata
+        .compilation_units
+        .iter()
+        .find(|m| m.package == package_id && m.target.kind == LIB_TARGET_KIND)
+}
+
+fn get_corelib(
+    compilation_unit_metadata: &CompilationUnitMetadata,
+) -> &CompilationUnitComponentMetadata {
+    compilation_unit_metadata
+        .components
+        .iter()
+        .find(|du| du.name == CORELIB_CRATE_NAME)
+        .expect("Corelib could not be found")
+}
+
+fn get_dependencies(
+    compilation_unit_metadata: &CompilationUnitMetadata,
+) -> OrderedHashMap<SmolStr, PathBuf> {
+    compilation_unit_metadata
+        .components
+        .iter()
+        .filter(|du| du.name != "core")
+        .map(|cu| {
+            (
+                cu.name.to_smolstr(),
+                cu.source_root().to_owned().into_std_path_buf(),
+            )
+        })
+        .collect()
+}
+
+fn get_crates_config(
+    metadata: &Metadata,
+    compilation_unit_metadata: &CompilationUnitMetadata,
+) -> AllCratesConfig {
+    let crates_config: OrderedHashMap<SmolStr, CrateSettings> = compilation_unit_metadata
+        .components
+        .iter()
+        .map(|component| {
+            let pkg = metadata.get_package(&component.package).unwrap_or_else(|| {
+                panic!(
+                    "Failed to find = {} package",
```

Comment:
I would suggest using scarb style of error messages here as well, so please don't start with capital letters 
```suggestion
                    "failed to find `{}` package",
```